src/test/java/com/github/mtakaki/credentialstorage/hibernate/UnitOfWorkApplicationListenerTest.java

```diff
@@ -0,0 +1,397 @@
+
+        verify(this.session).setCacheMode(CacheMode.IGNORE);
+    }
+
+    @Test
+    public void configuresTheSessionsFlushMode() throws Exception {
+        this.prepareAppEvent("methodWithFlushModeAlwaysAnnotation");
+
+        this.execute();
+
+        verify(this.session).setFlushMode(FlushMode.ALWAYS);
+    }
+
+    @Test
+    public void doesNotBeginATransactionIfNotTransactional() throws Exception {
+        final String resourceMethodName = "methodWithTransactionalFalseAnnotation";
+        this.prepareAppEvent(resourceMethodName);
+
+        when(this.session.getTransaction()).thenReturn(null);
+
+        this.execute();
+
+        verify(this.session, never()).beginTransaction();
+        verifyZeroInteractions(this.transaction);
+    }
+
+    @Test
+    public void detectsAnnotationOnHandlingMethod() throws NoSuchMethodException {
+        final String resourceMethodName = "handlingMethodAnnotated";
+        this.prepareAppEvent(resourceMethodName);
+
+        this.execute();
+
+        verify(this.session).setDefaultReadOnly(true);
+    }
+
+    @Test
+    public void detectsAnnotationOnDefinitionMethod() throws NoSuchMethodException {
+        final String resourceMethodName = "definitionMethodAnnotated";
+        this.prepareAppEvent(resourceMethodName);
+
+        this.execute();
+
+        verify(this.session).setDefaultReadOnly(true);
+    }
+
+    @Test
+    public void annotationOnDefinitionMethodOverridesHandlingMethod() throws NoSuchMethodException {
+        final String resourceMethodName = "bothMethodsAnnotated";
+        this.prepareAppEvent(resourceMethodName);
+
+        this.execute();
+
+        verify(this.session).setDefaultReadOnly(true);
+    }
+
+    @Test
+    public void beginsAndCommitsATransactionIfTransactional() throws Exception {
+        this.execute();
+
+        final InOrder inOrder = inOrder(this.session, this.transaction);
+        inOrder.verify(this.session).beginTransaction();
+        inOrder.verify(this.transaction).commit();
+        inOrder.verify(this.session).close();
+    }
+
+    @Test
+    public void rollsBackTheTransactionOnException() throws Exception {
+        this.executeWithException();
+
+        final InOrder inOrder = inOrder(this.session, this.transaction);
+        inOrder.verify(this.session).beginTransaction();
+        inOrder.verify(this.transaction).rollback();
+        inOrder.verify(this.session).close();
+    }
+
+    @Test
+    public void doesNotCommitAnInactiveTransaction() throws Exception {
+        when(this.transaction.isActive()).thenReturn(false);
+
+        this.execute();
+
+        verify(this.transaction, never()).commit();
+    }
+
+    @Test
+    public void doesNotCommitANullTransaction() throws Exception {
+        when(this.session.getTransaction()).thenReturn(null);
+
+        this.execute();
+
+        verify(this.transaction, never()).commit();
+    }
+
+    @Test
+    public void doesNotRollbackAnInactiveTransaction() throws Exception {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://www.codacy.com/app/mitsuotakaki/CredentialStorageService-dw-hibernate/file/2455877775/issues/source?bid=3237716&fileBranchId=3250434#l232)